Acceptance tests: every geni_document and geni_photo destroy check fails with "still exists"

## Summary

A full run of `TF_ACC=1 go test -v ./test/acceptance/ -timeout 60m` against the Geni sandbox finishes with **57 PASS / 18 FAIL**. **All 18 failures share the same shape**: the test successfully creates the resource and runs through every step, but the post-test `CheckDestroy` fixture in `helpers_test.go` (`testAccCheckDocumentDestroy` line 70, `testAccCheckPhotoDestroy` line 91) reads the resource back via `client.Document().Get(...)` / `client.Photo().Get(...)` and gets a 200 instead of `geni.ErrResourceNotFound` — so the test framework reports `Error running post-test destroy, there may be dangling resources: document <id> still exists` (or `photo <id>`).

This is **isolated to the `geni_document` and `geni_photo` resources**. `geni_profile`, `geni_union`, `geni_project`, and the data-source tests all pass — and they use the same transport / retry path, so this isn't a network or auth regression.

## Failing tests

geni_document (13):

- TestAccDocument_listResources
- TestAccDocument_createTextDocument
- TestAccDocument_createTextDocumentWithDetails
- TestAccDocument_updateTextDocumentWithDetails
- TestAccDocument_createPngDocument
- TestAccDocument_createPngDocumentWithDetails
- TestAccDocument_updatePngDocumentWithDetails
- TestAccDocument_createUrlDocument
- TestAccDocument_generateConfigOut
- TestAccDocument_createUrlDocumentWithDetails
- TestAccDocument_createUrlDocumentWithProfile
- TestAccDocument_updateUrlDocumentWithDetails
- TestAccDocument_updateUrlDocumentWithProfiles

geni_photo (5):

- TestAccPhoto_listResources
- TestAccPhoto_createPhoto
- TestAccPhoto_createPhotoWithDescription
- TestAccPhoto_updatePhoto
- TestAccPhoto_createPhotoTaggedWithProfile

The "import non-existent ID" negative-path tests pass for both resource types, since they never create anything.

## Likely root cause

One of:

1. Geni's sandbox no longer hard-deletes documents/photos — a `GET` after `DELETE` still returns 200 (possibly returning a soft-deleted record).
2. The `Delete` implementation in `go-geni`'s `document` / `photo` packages, or in the provider's `internal/resource/document` / `photo` deletion code, regressed and silently drops the request.
3. Geni's API now returns 200 with an empty body or a `deleted: true` flag instead of 404 for already-deleted resources — and the client doesn't translate that into `ErrResourceNotFound`.

CI (`.github/workflows/ci.yaml`) only runs `make test` (unit tests), so no signal would have surfaced this earlier. The failures are clearly **pre-existing** — they reproduced cleanly on top of the issue #122 retry classifier fix branch, and the retry change can only add retries, never affect `DELETE` semantics.

## Reproduction

\`\`\`
GENI_ACCESS_TOKEN=<sandbox token> TF_ACC=1 go test -v ./test/acceptance/ -run 'TestAccDocument_createTextDocument$' -timeout 5m
\`\`\`

Expected: PASS. Actual:

\`\`\`
geni_document_test.go:14: Error running post-test destroy, there may be dangling resources: document document-XXXXX still exists
--- FAIL: TestAccDocument_createTextDocument
\`\`\`

## Suggested investigation order

1. Manually `DELETE` a document via curl against the sandbox, then `GET` it — does Geni return 404, or 200 with a soft-delete marker?
2. If (1) returns 200, decide whether `Document.Get` in `go-geni` should map that shape onto `ErrResourceNotFound`, or whether `testAccCheckDocumentDestroy` should accept it as deleted.
3. Same flow for photos.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,17 @@
 ## 0.22.3 (Unreleased)
 
+BUG FIXES:
+
+* `resource/geni_document`, `resource/geni_photo`: a `terraform destroy`
+  now correctly observes the resource as gone on the next `Read`.
+  Previously Geni's empty bulk-envelope reply (`{"results": []}`) to a
+  singular `/api/<id>` GET against a deleted document or photo was
+  silently unmarshaled into a zero-valued struct, so the provider
+  treated the deleted resource as still present and reported drift
+  (and acceptance tests' `CheckDestroy` fixture reported "still
+  exists"). Fixed in go-geni v1.9.2 by translating that response shape
+  into `ErrResourceNotFound`. (#123)
+
 ## 0.22.2
 
 BUG FIXES:

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.26
 
 require (
 	github.com/allegro/bigcache/v3 v3.1.0
-	github.com/dmalch/go-geni v1.9.1
+	github.com/dmalch/go-geni v1.9.2
 	github.com/hashicorp/terraform-plugin-framework v1.19.0
 	github.com/hashicorp/terraform-plugin-framework-validators v0.19.0
 	github.com/hashicorp/terraform-plugin-go v0.31.0

